src/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.cpp

```diff
@@ -418,6 +503,231 @@ Sequence::Solution Sequence::solveToCondition(
+                }
+
+                case MaximumManeuverDurationStrategy::Skip:
+                {
+                    segmentToSolve = aSegment.toCoastSegment(
+                        aSegment.getName() + "(Coast - Skipping Maneuver due to Maximum Maneuver Duration Constraint)"
+                    );
+                    coastToInstant = candidateManeuver.getInterval().getEnd();
+                    continue;
+                }
+
+                case MaximumManeuverDurationStrategy::Slice:
+                {
+                    const Shared<Thruster> shortenedManeuverThruster = buildThrusterDynamicsWithManeuverIntervals_(
+                        aSegment,
+                        {Interval::Closed(
+                            candidateManeuver.getInterval().getStart(),
+                            candidateManeuver.getInterval().getStart() + maximumManeuverDuration_
+                        )},
+                        "(Sliced Maneuver)"
+                    );
+
+                    segmentToSolve = aSegment.toManeuverSegment(
+                        shortenedManeuverThruster, aSegment.getName() + " (Sliced Maneuver)"
+                    );
+                    coastToInstant = candidateManeuver.getInterval().getStart() + maximumManeuverDuration_;
+                    break;
+                }
+
+                case MaximumManeuverDurationStrategy::Center:
+                {
+                    const Shared<Thruster> shortenedManeuverThruster = buildThrusterDynamicsWithManeuverIntervals_(
+                        aSegment,
+                        {Interval::Centered(
+                            candidateManeuver.getInterval().getCenter(),
+                            maximumManeuverDuration_,
+                            Interval::Type::Closed
+                        )},
+                        "(Center Maneuver)"
+                    );
+                    segmentToSolve = aSegment.toManeuverSegment(
+                        shortenedManeuverThruster, aSegment.getName() + " (Centered Maneuver)"
+                    );
+                    coastToInstant = candidateManeuver.getInterval().getEnd();
+                    break;
+                }
+
+                default:
+                {
+                    throw ostk::core::error::runtime::ToBeImplemented("Maximum maneuver duration strategy");
+                }
+            }
+
+            continue;
+        }
+
+        const Array<State>& segmentStates =
+            Array<State>(maneuverSegmentSolution.states.begin() + 1, maneuverSegmentSolution.states.end());
+        states.add(segmentStates);
+        maneuverIntervals.add(candidateManeuver.getInterval());
+        heterogeneousGuidanceLaw->addGuidanceLaw(
+            std::const_pointer_cast<GuidanceLaw>(maneuverSegmentSolution.getThrusterDynamics()->getGuidanceLaw()),
+            candidateManeuver.getInterval()
+        );
+
+        if (conditionIsSatisfied)
+        {
+            break;
+        }
```

Comment:
It seems that this statement is placed incorrectly in the code? It should go right after somewhere that changes its value, like on line 565